src/BaGet.Core/PackageDatabase.cs

```diff
@@ -11,24 +11,28 @@ namespace BaGet.Core
     public class PackageDatabase : IPackageDatabase
     {
         private readonly IContext _context;
+        private readonly Func<IContext> _newContext;
 
-        public PackageDatabase(IContext context)
+        public PackageDatabase(IContext context, Func<IContext> newContext)
```

Comment:
Suggest naming this context generator so it's clearer what it is?

Reply:
I prefer to name the factory closer to the idiomatic C# code, i.e. 

```
using var context =_newContext();
```
as an analog to
```
using var context = new Context();
```

I can can probably be convinced to rename to _createContext ;)

Let's be honest though, we all nab the fixes and then apply our own naming conventions here.